### PR TITLE
Add missing tests for block_create

### LINKED ISSUE #6462: Server test block_create

### PR DESCRIPTION
**Pull request by mmmarcos**
Closes #6462

### CODE REVIEW

File: server/tests/api_v4/authenticated/test_block_create.py [hunk 1]
```diff
@@ -27,6 +31,28 @@ async def test_authenticated_block_create_ok(coolorg: CoolorgRpcClients, backend
...
+) -> None:
+    block_id = BlockID.new()
+    block = b"<block content>"
+
+    # Try to create block with an invalid realm ID
+    rep = await coolorg.alice.block_create(block_id, VlobID.new(), block)
+    assert rep == authenticated_cmds.v4.block_create.RepRealmNotFound()
+
+    # Block should not exist
+    content = await backend.block.read_as_user(
```
Comment (touilleMan) [inline server/tests/api_v4/authenticated/test_block_create.py:45]: No need for this part given `test_dump_blocks` already shows us the block doesn't exist

Note: `read_as_user` is present for the `_ok`  test case. This is not strictly required, but works as a cheap check to ensure the block can be actually retrieved (while testing the block cannot be retrieved when it is not present in the first place seems very paranoid).

In the end I'm fine with either of those solutions:
- do nothing and keep `read_as_user` check in all tests
- considering `test_dump_blocks` is already enough and remove `read_as_user` in all tests 
- keep `read_as_user` only in the `_ok` test to balance test readability

your call :+1:

File: server/tests/api_v4/authenticated/test_block_create.py [hunk 2]
```diff
@@ -27,6 +31,28 @@ async def test_authenticated_block_create_ok(coolorg: CoolorgRpcClients, backend
...
+        organization_id=coolorg.organization_id,
+        author=coolorg.alice.device_id.user_id,
+        block_id=block_id,
+    )
+    assert content != BlockReadBadOutcome.BLOCK_NOT_FOUND.value
```
Comment (touilleMan) [inline server/tests/api_v4/authenticated/test_block_create.py:50]: This seems strange to me: `read_as_user` is supposed to return a `BlockReadBadOutcome` enum, and we expect it to be abound the block not being found...

```suggestion
    assert content = BlockReadBadOutcome.BLOCK_NOT_FOUND
```

File: server/tests/api_v4/authenticated/test_block_create.py [hunk 3]
```diff
@@ -63,4 +89,32 @@ async def test_authenticated_block_create_not_allowed(
...
+    rep = await coolorg.alice.block_create(block_id, coolorg.wksp1_id, block)
+    assert rep == authenticated_cmds.v4.block_create.RepStoreUnavailable()
+
+    # Block should not exist
+    content = await backend.block.read_as_user(
+        organization_id=coolorg.organization_id,
+        author=coolorg.alice.device_id.user_id,
+        block_id=block_id,
+    )
+    assert content != BlockReadBadOutcome.BLOCK_NOT_FOUND.value
```
Comment (touilleMan) [inline server/tests/api_v4/authenticated/test_block_create.py:117]: same as above
```suggestion
    assert content = BlockReadBadOutcome.BLOCK_NOT_FOUND
```

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: server/tests/api_v4/authenticated/test_block_create.py (hunk 1 above)
Comment (mmmarcos) [reply server/tests/api_v4/authenticated/test_block_create.py:48]: You're right. Actually, if ever `read_as_user` check fails, but not `test_dump_blocks`, that would mean there is a bug in `read_as_user`, not really in `block_create`.

Anyway, maybe it is better to leave it as a "cheap check". I added some comments based on your review.
WDYT?

Comment (touilleMan) [reply server/tests/api_v4/authenticated/test_block_create.py:48]: > WDYT

:+1:

**Pull request merged**